app/views/play/cinematic/Loader.js

```diff
@@ -27,45 +26,97 @@ export default class Loader {
    * @returns {Cinematic} The raw JSON object with Cinematic data.
    */
   async loadAssets () {
-    this.data = await get(this.slug)
     this.loadThangTypes(this.data.shots)
+    this.loadPlayerThangType()
+    this.loadBackgroundObjects(this.data.shots)
     await this.load()
     return this.data
   }
 
+  /**
+   * Loads the player thangType from the global `me` object if accessible.
+   * Has a side effect of storing the players thangType by original as a resource.
+   */
+  loadPlayerThangType () {
+    if ((me || {}) && !me.get('heroConfig')) {
+      return
+    }
+    const original = me.get('heroConfig').thangType
+    if (!original) {
+      return
+    }
+
+    this.loadingThangTypes.set(
+      original,
+      (async () => {
+        const attr = await getThangTypeOriginal(original)
+        this.loadedThangTypes.set(original, new ThangType(attr))
+      })()
+    )
+  }
+
+  /**
+   * Queues up the background objects
+   * @param {Shot[]} shots
+   */
+  loadBackgroundObjects (shots) {
+    shots
+      .filter(shot => (shot.dialogNodes || []).length > 0)
+      .map(({ dialogNodes }) =>
```

Comment:
Should this be `forEach` to be consistent with line 71?

Reply:
It definitely should be. Thanks for mentioning this. It's good to use the method with the right intention behind it.